kalite/distributed/features/steps/login.py

```diff
@@ -0,0 +1,118 @@
+from behave import *
+from selenium.common.exceptions import NoSuchElementException
+
+from django.core.urlresolvers import reverse
+
+from kalite.testing.behave_helpers import *
+
+from kalite.facility.models import Facility
+
+from kalite.testing.mixins.facility_mixins import CreateFacilityMixin, CreateStudentMixin
+
+TIMEOUT = 10
+
+@given("there is one facility")
+def step_impl(context):
+    if Facility.objects.count() != 0:
+        for f in Facility.objects.all():
+            f.soft_delete()
+    context.facility = CreateFacilityMixin.create_facility()
+
+@given("there is more than one facility")
+def step_impl(context):
+    name = 0
+    while Facility.objects.count() < 2:
+        context.facility = CreateFacilityMixin.create_facility(name=str(name))
+        name += 1
+
+@given("I am on the homepage")
+def step_impl(context):
+    go_to_homepage(context)
+
+def go_to_homepage(context):
+    url = reverse("homepage")
+    context.browser.get(build_url(context, url))
+
+@when("I click log in")
+def step_impl(context):
+    nav_login = find_id_with_wait(context, "nav_login")
+    click_and_wait_for_id_to_appear(context, nav_login, "id_username-container")
+
+
+@then("there should be no facility drop down")
+def step_impl(context):
+    try:
+        assert not find_id_with_wait(context, "id_facility-container")
+    except NoSuchElementException:
+        assert True
+
+@then("there should be a facility drop down")
+def step_impl(context):
+    assert find_id_with_wait(context, "id_facility-container")
+
+@given('I have an account')
+def impl(context):
+    context.user = CreateStudentMixin.create_student()
+    context.password = CreateStudentMixin.DEFAULTS["password"]
+
+@when('I enter my password incorrectly')
+def impl(context):
+    fill_password(context, "notarealpasswordIpromise")
+
+@when('I enter my password correctly')
+def impl(context):
+    fill_password(context, context.password)
+
+@when('I enter my username incorrectly')
+def impl(context):
+    fill_username(context, "notarealusernameIswear")
+
+@when('I enter my username correctly')
+def impl(context):
+    fill_username(context, context.user.username)
+
+@when('I click the login button')
+def impl(context):
+    login_button = find_css_class_with_wait(context, "login-btn")
+    login_button.click()
```

Comment:
Okay, upon closer inspection I think one possibility is that the step which follows this one (`@then('the page should reload')`) is subject to a race condition if the new page loads before Selenium grabs an element from the current page.